tests/components/opengarage/test_config_flow.py

```diff
@@ -0,0 +1,101 @@
+"""Test the OpenGarage config flow."""
+import aiohttp
+
+from homeassistant import config_entries, setup
+from homeassistant.components.opengarage.const import DOMAIN
+
+from tests.async_mock import patch
+from tests.common import MockConfigEntry
+
+
+async def test_form(hass):
+    """Test we get the form."""
+    await setup.async_setup_component(hass, "persistent_notification", {})
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": config_entries.SOURCE_USER}
+    )
+    assert result["type"] == "form"
+    assert result["errors"] == {}
+
+    with patch(
+        "opengarage.OpenGarage.update_state",
+        return_value={"name": "Name of the device", "mac": "unique"},
+    ), patch(
+        "homeassistant.components.opengarage.async_setup", return_value=True
+    ) as mock_setup, patch(
+        "homeassistant.components.opengarage.async_setup_entry", return_value=True,
+    ) as mock_setup_entry:
+        result2 = await hass.config_entries.flow.async_configure(
+            result["flow_id"], {"host": "http://1.1.1.1", "device_key": "AfsasdnfkjDD"},
+        )
+
+    assert result2["type"] == "create_entry"
+    assert result2["title"] == "Name of the device"
+    assert result2["data"] == {
+        "host": "http://1.1.1.1",
+        "device_key": "AfsasdnfkjDD",
+        "port": 80,
+        "verify_ssl": False,
+    }
+    await hass.async_block_till_done()
+    assert len(mock_setup.mock_calls) == 1
+    assert len(mock_setup_entry.mock_calls) == 1
+
+
+async def test_form_cannot_connect(hass):
+    """Test we handle cannot connect error."""
+    result = await hass.config_entries.flow.async_init(
```

Comment:
We can pass user input directly to `flow.async_init` as the `data` parameter. Ie use that below instead of `flow.async_configure`.

Reply:
Do you have an example for that?

Reply:
https://github.com/home-assistant/core/blob/e6065569ae1e3817b6e539c4a72be71acc92380b/tests/components/plex/test_config_flow.py#L83-L91